app/controllers/questions_controller.rb

```diff
@@ -19,9 +23,111 @@ def show
       gon.solution = solution
     end
 
+    if !(current_user.show_solutions)
+      if @room.answered(@question.id,@room.id)
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id) + 1
+      else
+        @number_of_question = @room.question_count - @room.question_count_not_visited(@room.id)
```

Comment:
Ten kod je vo vetve `if` aj v `else` rovnaky az na to `+ 1`. Treba dodrziavat princip DRY(Don't Repeat Yourself), napr. ked bude toto potrebne zmenit, meni sa to na dvoch miestach. Cize lepsie je najskor jednym prikazom priradit do premennej tu spolocnu cast a potom podla potreby este ne/priratat tu jednotku.